ml-agents/mlagents/trainers/policy/tf_policy.py

```diff
@@ -392,18 +400,50 @@ def get_update_vars(self):
         """
         return list(self.update_dict.keys())
 
-    def save_model(self, steps):
+    def checkpoint(self, model_reward: Optional[float] = None) -> None:
```

Comment:
Personally I think this should stay as `save_model(self, steps: int)` as "steps" is something the trainer know, not necessarily the policy - the fact that the policy knows what step it's on is more of an artifact of wanting to save/resume that variable from the TF checkpoint. Moving forward (e.g. in PyTorch) we're likely to move saving steps in the `training_status.json` and not in the policy.

Reply:
@ervteng `policy.get_current_step()` is used in several places in any case at the moment -- maybe we can cross this bridge when we come to it?

I think the idea of needing to pass in a bunch of things from the Trainer just so a Policy can checkpoint is also not the best abstraction.  Seems like we may just want a `Policy.checkpoint()` and `Policy.save(path: str)`, and have the Trainer manage the checkpoints with the CheckpointManager?

Reply:
As far as I can tell `get_current_step()` is only used to load the past steps from the checkpoint - if we ran it at any frequency during training it would be quite slow 😛 . 

I think having a `Policy.save(path:str)` would work as then the Policy doesn't also have to worry about the model directory and so forth. But where would `Policy.checkpoint()` save to?

Reply:
Added back steps for now, let's keep in mind further separating the Policy from knowing about these things in the future.